fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -109,6 +119,18 @@ public boolean isAllowNull() {
         return isAllowNull;
     }
 
+    // For database null constraint, for all key models, there are three choices in starrocks for user now(until 2.0 deb29af76ee749ed8e99ec002cce4881b507c230):
+    // 1. explicit specify, null
+    // 2. explicit specify, not null
+    // 3. implicit specified, that is write nothing, the default behavior is null
+    // Product manager @Dshadowzh what to do #2778 to make primary key set NOT NULL by default in the 3 implicit specified scenario.
+    // so, according to this proposal, for primary key, should use setPrimaryKeyNonNullable() to update the real value to `isAllowNull` in create table stage, if the null constraint was implicit specified, should set to false, which is not null.
```

Comment:
So add a new setPrimaryKeyNonNullable() to set `isAllowNull` to be true for primary key columns.